addons/mail/static/src/discuss/discuss.js

```diff
@@ -16,9 +16,30 @@ export class Discuss extends Component {
 
     setup() {
         this.messaging = useMessaging();
+        this.state = useState({
+            doFocusOnThreadNameInput: false,
+            isEditingThreadName: false,
+            isMouseOverThreadName: false,
+            pendingThreadName: "",
+        });
+        this.threadNameInputRef = useRef("threadNameInput");
+        this._onClickCaptureGlobal = this._onClickCaptureGlobal.bind(this);
+
         onWillStart(() => this.messaging.isReady);
-        onMounted(() => (this.messaging.discuss.isActive = true));
-        onWillUnmount(() => (this.messaging.discuss.isActive = false));
+        onMounted(() => {
+            this.messaging.discuss.isActive = true;
+            document.addEventListener('click', this._onClickCaptureGlobal, true);
+        });
+        onPatched(() => {
+            if (this.state.doFocusOnThreadNameInput && this.threadNameInputRef.el) {
+                this.threadNameInputRef.el.focus();
+                this.state.doFocusOnThreadNameInput = false;
+            }
+        });
+        onWillUnmount(() => {
+            this.messaging.discuss.isActive = false;
+            document.removeEventListener('click', this._onClickCaptureGlobal, true);
```

Comment:
For `_onClickCaptureGlobal`, you can use `useExternalListener('document', 'click', this._onClickCaptureGlobal, { capture: true })`